dotcom-rendering/src/lib/fetchEmail.ts

```diff
@@ -0,0 +1,35 @@
+import type { IdApiUserData } from './getIdapiUserData';
+import { getIdApiUserData } from './getIdapiUserData';
+import { eitherInOktaTestOrElse } from './identity';
+
+const getEmail = async (ajaxUrl: string): Promise<string | undefined> =>
+	// TODO Okta: Remove either when at 100% in oktaVariant test, and just use idToken
+	eitherInOktaTestOrElse(
+		(authState) => authState.idToken?.claims.email,
+		() =>
+			getIdApiUserData(ajaxUrl)
+				.then((data: IdApiUserData) => data.user?.primaryEmailAddress)
+				.catch((error) => {
+					window.guardian.modules.sentry.reportError(
+						error,
+						'getEmail',
+					);
+					return undefined;
+				}),
+	);
+
+export const lazyFetchEmailWithTimeout =
+	(idapiUrl: string): (() => Promise<string | null>) =>
+	() => {
+		return new Promise((resolve) => {
+			setTimeout(() => resolve(null), 1000);
+			// eslint-disable-next-line @typescript-eslint/no-floating-promises
+			getEmail(idapiUrl).then((email) => {
```

Comment:
The ESLint rules exist for a reason, [which in this case is ensuring that a rejection is handled](https://typescript-eslint.io/rules/no-floating-promises/)!

```suggestion
			void getEmail(idapiUrl).then((email) => {
```